server/src/cache.js

```diff
@@ -0,0 +1,39 @@
+const storage = {}
+
+/**
+ * This function will access the cache and return its content if it is a hit.
+ * If it is a miss it will use the provided function to calculate the value.
+ *
+ * The nature of the method is to always return the cache's content,
+ * if there is any and only update itself after. This means it can return
+ * out-dated data.
+ *
+ * @param {Function} func a long running async function,
+ *  it will be used to update the cache with itemKeys as an argument
+ * @param {String} cacheKey the key under which the cache will be stored
+ * @param {Array} itemKeys the underlying items to be accessed
+ */
+export async function getAndUpdateCache(func, cacheKey, itemKeys) {
```

Comment:
There is no reason why this should be so complicated.

```js
export async function create(load) {
  return {
    storage: Object.create(null),
    load,
  }
}

export async function get(cache, key) {
  const update = cache.load(key).then((v) => {cache.storage[key] = v})
  if (cache.storage[key] === undefined) await update
  return cache.storage[key]
}

Reply:
This looks definitely better. I have overthought it.